### ISSUE TITLE
Make diff window load asynchronously

### ISSUE THREAD

**Issue by atimholt** [mbbill/undotree#104]
On very big diffs (e.g. an external-tool changes the whole file/buffer on a very long file, e.g. reformatting the code), the entire editor can freeze for several minutes. Perhaps a “loading diff” message could be displayed until it is finished loading it.

I’m running into this problem when trying to open undotree in a clang-formatted version of vulkan.hpp, an 80,000 line file (saved as a different file, of course). I open it for reference, but want it to be more readable, so I use an external C++ formatter. Then, when I want to glance at my undo history, to make sure I haven’t done anything else, Vim freezes for ~5 minutes.

Yeah, I know, I need to set the buffer to `noma ro` and probably manually delete my undo history once I’ve got it how I want it. It’d be nice to not have to lose my history, though. Also, I’m sure there’s plenty of cases where someone still needs to edit the file. They might be *bad* use cases, but it’s not up to undotree to punish bad coding practices with long waits :).

**Comment by mbbill** [comment]
Yeah sometimes I run into this kind of situation too, even on some files with only couple thousands of lines I can still notice the lag. The plugin was written when async scripting wasn't a thing, so it's not quite prepared for that. Let's see if I can hack a way through it.

**Comment by atimholt** [comment]
Also, this may be obvious, but make sure to terminate the diff-window process when undotree is toggled off, if possible. I’m guessing it’s probably a batch process without built-in interruption, but maybe evaluate whether a hard terminate has any side effects.

**Comment by atimholt** [comment]
Some further research shows that (maybe?) *Neovim’s* lua engine supports multithreading, but they’re using (or will be using?) something called “libuv” for that, a cross-platform C library for async.
